frontend/src/routes/manage/Realm/Content/Edit/EditMode/index.tsx

```diff
@@ -0,0 +1,174 @@
+            id
+            blocks {
+                id
+                title
+                __typename
+                ... on TextBlock { ... TextEditModeBlockData }
+                ... on SeriesBlock { ... SeriesEditModeBlockData }
+            }
+        }
+    `, realmRef);
+    const block = blocks[index];
+    const { id, title, __typename: type } = block;
+
+
+    const form = useForm<EditModeFormData>({
+        defaultValues: {
+            type: type as "TextBlock" | "SeriesBlock",
+        },
+    });
+    const editModeRef = useRef<EditModeRef>(null);
+
+    const onSubmit = form.handleSubmit(data => {
+        // Empty titles should set the field to `null`
+        // This is to avoid empty headings when rendering the block.
+        // In the future we might want to check this at render site,
+        // and we also might want more sophisticated checks
+        // (like "all whitespace").
+        const processData = { ...data, title: data.title || null };
+
+        const currentEditModeRef = editModeRef.current ?? bug("`editModeRef` not bound");
+
+        onSave?.();
+
+        if (id.startsWith("cl")) {
+            currentEditModeRef.create(
+                realmId,
+                index,
+                processData,
+                onCompleted,
+                onError,
+            );
+        } else {
+            currentEditModeRef.save(
+                id,
+                processData,
+                onCompleted,
+                onError,
+            );
+        }
+    });
+
+
+    return <FormProvider<EditModeFormData> {...form}>
+        <form onSubmit={onSubmit}>
+            <EditModeButtons onCancel={onCancel} />
+            <h2 css={{ margin: "16px 0" }}>
+                <Input
+                    css={{ display: "block" }}
+                    placeholder={t("manage.realm.content.title")}
+                    defaultValue={title ?? ""}
+                    {...form.register("title")}
+                />
+            </h2>
+            {match(block.__typename, {
+                "TextBlock": () => <EditTextBlock ref={editModeRef} block={block} />,
+                "SeriesBlock": () => <EditSeriesBlock ref={editModeRef} block={block} />,
+            })}
+        </form>
+    </FormProvider>;
+};
+
+
+type EditModeButtonsProps = {
+    onCancel?: () => void;
+};
+
+const EditModeButtons: React.FC<EditModeButtonsProps> = ({ onCancel }) => {
+    const { t } = useTranslation();
+
+    const modalRef = useRef<ConfirmationModalHandle>(null);
+
+    return <ButtonGroup css={{ marginTop: -24 }}>
+        <Button
+            title={t("manage.realm.content.cancel")}
+            onClick={() => modalRef.current?.open()}
+        >
+            <FiX />
+        </Button>
+        <Button
+            type="submit"
+            title={t("manage.realm.content.save")}
+        >
+            <FiCheck />
+        </Button>
+        <ConfirmationModal
+            title={t("manage.are-you-sure")}
```

Comment:
I've seen this line lots of times already. Maybe make that the default value of `ConfirmationModal` or sth...

Reply:
Although that would make `ConfirmationModal` depend on the `i18n` stuff, which it currently does not. I mean that's a weak argument, admittedly, but that also means it can't just be a default argument, because we don't have access to the translation function at definition time. Or we could pass the key? :thinking: 

Or we just hard code it in `ConfirmationModal`.

Opinions? :sweat_smile:

Reply:
Hard code or make the prop optional and do `title ?? t("...)"` somewhere inside that component.

Reply:
Done the latter in f3ed39a.